internal/liqonet/tunnel-operator/tunnel-operator.go

```diff
@@ -488,144 +422,76 @@ func (tc *TunnelController) SetUpRouteManager() error {
 	return nil
 }
 
-func (tc *TunnelController) setUpGWNetns(netnsName, hostVethName, gatewayVethName, gatewayVethIPAddr string, vethMtu int) error {
+func (tc *TunnelController) setUpGWNetns(hostVethName, gatewayVethName string, vethMtu int) error {
 	var err error
+
 	// Create veth pair to connect the two namespaces.
-	err = liqonetns.CreateVethPair(hostVethName, gatewayVethName, tc.hostNetns, tc.gatewayNetns, vethMtu)
+	hostVeth, gatewayVeth, err := liqonetns.CreateVethPair(hostVethName, gatewayVethName, tc.hostNetns, tc.gatewayNetns, vethMtu)
 	if err != nil {
-		klog.Errorf("an error occurred while creating the veth pair: %s", err)
 		return err
 	}
-	klog.Infof("created veth device {%s} in host network", hostVethName)
-	klog.Infof("created veth device {%s} in gateway custorm network namespace {%s}", gatewayVethName, netnsName)
-	if err = tc.configureGatewayNetns(gatewayVethName, gatewayVethIPAddr, tc.gatewayNetns); err != nil {
+	klog.Infof("created veth device {%s} with index {%d} in host netns with path {%s}",
+		hostVeth.Name, hostVeth.Index, tc.hostNetns.Path())
+	klog.Infof("created veth device {%s} with index {%d} in gateway netns with path {%s}",
+		gatewayVeth.Name, gatewayVeth.Index, tc.gatewayNetns.Path())
+
+	tc.hostVeth = hostVeth
+	tc.gatewayVeth = gatewayVeth
+
+	// Configure hostveth.
+	if err = liqonetns.ConfigureVeth(&hostVeth, liqoconst.GatewayVethIPAddr, gatewayVeth.HardwareAddr, tc.hostNetns); err != nil {
 		return err
 	}
-	// Enable arp proxy for liqo.host veth interface.
-	var configureHostNetns = func(netNamespace ns.NetNS) error {
-		if err := liqorouting.EnableProxyArp(hostVethName); err != nil {
-			return err
-		}
-		klog.Infof("enabled proxy arp for veth device {%s}", hostVethName)
-		// Get veth interface.
-		veth, err := netlink.LinkByName(hostVethName)
-		if err != nil {
-			return err
-		}
-		tc.hostVeth = veth
-		if _, err := liqorouting.AddRoute(gatewayVethIPAddr, "", veth.Attrs().Index, 0); err != nil {
-			return err
-		}
-		klog.Infof("added route for destination {%s} on device {%s}", gatewayVethIPAddr, hostVethName)
+
+	// Configure gatewayveth.
+	return liqonetns.ConfigureVeth(&gatewayVeth, liqoconst.HostVethIPAddr, hostVeth.HardwareAddr, tc.gatewayNetns)
+}
+
+func (tc *TunnelController) updateStatus(con *netv1alpha1.Connection, tep *netv1alpha1.TunnelEndpoint) error {
+	if reflect.DeepEqual(*con, tep.Status.Connection) && tep.Status.GatewayIP == tc.podIP &&
+		tep.Status.VethIFaceIndex == tc.hostVeth.Index && tep.Status.VethIP == liqoconst.GatewayVethIPAddr {
 		return nil
 	}
-	if err = tc.hostNetns.Do(configureHostNetns); err != nil {
+
+	tep.Status.Connection = *con
+	tep.Status.GatewayIP = tc.podIP
+	tep.Status.VethIFaceIndex = tc.hostVeth.Index
+	tep.Status.VethIFaceName = tc.hostVeth.Name
+	tep.Status.VethIP = liqoconst.GatewayVethIPAddr
```

Comment:
Wondering whether it is necessary to store this as part of the status, given it is always assigned a constant value.

Reply:
That's true but there are two reasons while I moved it to the status:

1. I would like to avoid the usage of constants values in functions that configure the routes for a given remote cluster;
2. I would not exclude that in the near future we will support separate VPN tunnel for each remote cluster.